draft-sharabayko-mops-srt.md

```diff
@@ -2086,15 +2086,584 @@ and RTTVar are updated by the socket based on algorithms for the sender
+~~~
+
+Step 3. Set the congestion window size to:
+
+~~~
+CWND_SIZE = RECEIVING_RATE * (RTT + RC_INTERVAL) / 1000000 + 16
+~~~
+
+Step 4. If there is packet loss reported by the receiver (bLoss=True):
+
+a. Keep the value of PKT_SND_PERIOD at the same level;
+
+b. Set the value of bLoss to False;
+
+c. Stop.
+
+bLoss flag is equal to True if a packet loss has happened since the
+last sending rate increase. Initial value: False.
+
+Step 5. If there is no packet loss reported by the receiver (bLoss=False),
+calculate PKT_SND_PERIOD as follows:
 
-Given that SRT can operate in live and file transfer modes, there are two groups of congestion control algorithms possible.
+    <CODE BEGINS>
+    inc = 0;
+
+    lossBandwidth = 2 * (1000000 / LastDecPeriod);
+    linkCapacity = min(lossBandwidth, EST_LINK_CAPACITY);
+    B = linkCapacity - 1000000 / PKT_SND_PERIOD;
+
+    if ((PKT_SND_PERIOD > LastDecPeriod) && ((linkCapacity / 9) < B))
+        B = linkCapacity / 9;
+    if (B <= 0)
+        inc = 1 / S;
+    else
+    {
+        inc = pow(10.0, ceil(log10(B * S * 8))) * 0.0000015 / S;
+        inc = max(inc, 1 / S);
+    }
+
+    PKT_SND_PERIOD = (PKT_SND_PERIOD * RC_INTERVAL) / 
+                      (PKT_SND_PERIOD * inc + RC_INTERVAL);
+    <CODE ENDS>
+
+where
+
+- LastDecPeriod is the value of PKT_SND_PERIOD right before the last
+sending rate decrease has happened (on a loss report (NAK) packet
+reception), in microseconds. The initial value of LastDecPeriod
+is set to 1 microsecond;
+- EST_LINK_CAPACITY is the estimated link capacity reported
+by the receiver within an ACK packet and smoothed at the sender side
+({{bandwidth-estimation}), in packets per second;
+- B is the estimated available bandwidth, in packets per second;
+- S is the SRT packet size (in terms of IP payload) in bytes.
+SRT treats 1500 bytes as a standard packet size.
+
+The detailed explanation of the formulas used to calculate the increase
+in sending rate inc can be found in {{GuAnAO}}. The UDT's available
+bandwidth B estimation has been modified to take into account the
+bandwidth registered at the moment of packet loss since the estimated
+link capacity EST_LINK_CAPACITY reported by the receiver may
+overestimate the real one significantly.
+
+Step 6. If the value of maximum bandwidth MAX_BW defined in
+{{liveCC}} is set, limit the value of PKT_SND_PERIOD to the
+minimum allowed period, if necessary:
+
+    <CODE BEGINS>
+    if (MAX_BW)
+        MIN_PERIOD = 1000000 / (MAX_BW / S);
+
+        if (PKT_SND_PERIOD < MIN_PERIOD)
+            PKT_SND_PERIOD = MIN_PERIOD;
+    <CODE ENDS> 
+
+Note that in file transmission mode there is an opportunity to define
+the maximum allowed bandwidth MAX_BW for SRT which limits the minimum
+possible packets' inter-sending period. Only the usage of MAXBW mode
+is possible (see {{maxbw}}). In contrast with live transmission
+mode, there is no default value set for MAX_BW and transmission rate
+is not limited if not set explicitly.
```

Comment:
```suggestion
Note that in file transmission mode the the maximum allowed bandwidth (MAX_BW) 
for SRT can be defined. This  limits the minimum possible interval between packets 
sent. Only the usage of MAXBW mode is possible (see {{maxbw}}). In contrast with 
live transmission mode, there is no default value set for MAX_BW, and the transmission 
rate is not limited if not set explicitly.
```

Reply:
Note that in file transmission mode the maximum allowed bandwidth (MAX_BW)
for SRT can be defined. This limits the minimum possible **packets' inter-sending period**.
Only the usage of MAXBW mode is possible ({{maxbw}}). In contrast with
live transmission mode, there is no default value set for MAX_BW, and the transmission
rate is not limited if not set explicitly.

I need to say "inter-sending period" here.

Reply:
Okay. Just so that you know "inter-sending" is not really a word!  ;-)

Also, in your comment above, you said "...we have two modes in SRT: message and buffer, ...". So maybe change "Note that in file transmission mode..." to "During file transmission..."?

Reply:
Okay, interesting. I know that there is a word "inter-arrival" ))) Will use you original suggestion + correct live/file modes.